syntax = "proto3";


option go_package = "./;pb";


message DB_PING_REQ
{
    int64   Timestamp = 1;
};

message DB_PING_RESP
{
    int64   TimeStamp = 1;
};

message DB_HEART_BEAT_REQ
{
    int64   Timestamp = 1;  // 时间戳
};

message DB_HEART_BEAT_RESP
{
    int64   Timestamp = 1;  // 时间戳
};


//////////////////////////////////
//////////////////////////////////
//////////////////////////////////

message CommReqData
{
    int32 ServerId = 1; // 服务器id
};

message CommRespData
{
    bool    Status  = 1; // 响应状态
    string  Msg     = 2; // 响应信息
};

message RedisConfig
{
    string  RedisIP     = 1;
    int32   RedisPort   = 2;
};

message MysqlConfig
{
    string  IP     = 1;
    int32   Port   = 2;
    string  Username = 3;
    string  Password = 4;
    string  DBName  = 5;
};

message RoleData
{
    string          RoleName        = 2;
    string          Account         = 3;
    string          EquipBaseData   = 4;
    string          EquipExtData    = 5;
    string          ItemData        = 6;
    string          PetData         = 7;
};

//////////////////////////////////

/* 初始化服务配置 */
message DB_INIT_DYNAIMC_CONFIG_REQ
{
    RedisConfig RedisCfg = 1;   // redis connect info
    MysqlConfig MysqlCfg = 2;   // mysql connect info
};

message DB_INIT_DYNAMIC_CONFIG_RESP
{
    CommRespData    CommData          = 1;
    bool            RedisConnInitSucc = 2;  // db service connect redis succ
    bool            MysqlConnInitSucc = 3;  // db service connect mysql succ
};                     

/* 取所有玩家密码 */
message DB_GET_ALL_ROLE_PWD_REQ
{
    CommReqData     CommData    = 1;
};

message DB_GET_ALL_ROLE_PWD_RESP
{
    repeated string          Pwd = 1;
};

/* 创建一个新角色 */
message DB_CREATE_ROLE_REQ
{
    CommReqData     CommData        = 1;
    RoleData        RoleData        = 2;
};

message DB_CREATE_ROLE_RESP
{
    CommRespData    CommData    = 1;
};


/* 请求角色数据 */
message DB_LOAD_ROLE_DATA_REQ
{
    CommReqData     CommData    = 1;
    int32           RoleId      = 2;
};

message DB_LOAD_ROLE_DATA_RESP
{
    CommRespData    CommData    = 1;
    RoleData        RoleData    = 2;
};